tests/Rector/Architecture/DoctrineRepositoryAsService/Source/RepositoryForDoctrineEntityProvider.php

```diff
@@ -0,0 +1,44 @@
+<?php declare(strict_types=1);
+
+namespace Rector\Tests\Rector\Architecture\DoctrineRepositoryAsService\Source;
+
+use Rector\Contract\Bridge\RepositoryForDoctrineEntityProviderInterface;
+
+final class RepositoryForDoctrineEntityProvider implements RepositoryForDoctrineEntityProviderInterface
+{
+    /**
+     * @var string[]
+     */
+    private $map = [
+        'AppBundle\Entity\Post' => 'AppBundle\Repository\PostRepository',
+    ];
+
+    public function provideRepositoryForEntity(string $name): ?string
+    {
+        if ($this->isAlias($name)) {
+            return $this->resoleFromAlias($name);
+        }
+
+        return $this->map[$name] ?? null;
+    }
+
+    private function isAlias(string $name): bool
+    {
+        return strpos($name, ':') !== false;
+    }
+
+    private function resoleFromAlias(string $name): ?string
+    {
+        [$namespaceAlias, $simpleClassName] = explode(':', $name, 2);
+
+        $pattern = sprintf('/(%s{1}.*%s)/', $namespaceAlias, $simpleClassName);
+
+        foreach ($this->map as $key => $value) {
+            if (preg_match($pattern, $key)) {
+                return $value;
+            }
+        }
+
+        return null;
+    }
```

Comment:
Having tests for this would make much easier to understand what this does for which inputs.
Now I have to think a lot and guess.

Could you add tests please?

Reply:
Oh, I see now. It's just part of a tests.

What is added value to have such complext dummy class only for test?

Reply:
It is to get repository from aliased parameter, what do you propose? Maybe move it to abstract class so it is part of any Provider?

Reply:
Abstraction seems good